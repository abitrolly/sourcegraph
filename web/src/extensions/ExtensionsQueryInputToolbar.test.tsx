import React from 'react'
import { EXTENSION_CATEGORIES } from '../../../shared/src/schema/extensionSchema'
import { extensionsQuery } from './extension/extension'
import { ExtensionsQueryInputToolbar } from './ExtensionsQueryInputToolbar'
import { mount } from 'enzyme'

describe('ExtensionsQueryInputToolbar', () => {
    test('renders', () => {
        expect(mount(<ExtensionsQueryInputToolbar query="q" onQueryChange={() => undefined} />)).toMatchSnapshot()
    })

    test('shows category in query as selected', () => {
        expect(
            mount(
                <ExtensionsQueryInputToolbar
                    query={extensionsQuery({ category: EXTENSION_CATEGORIES[0] })}
                    onQueryChange={() => undefined}
                />
            )
        ).toMatchSnapshot()
    })
})
