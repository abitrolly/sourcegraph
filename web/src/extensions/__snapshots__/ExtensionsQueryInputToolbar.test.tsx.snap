// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtensionsQueryInputToolbar renders 1`] = `
<ExtensionsQueryInputToolbar
  onQueryChange={[Function]}
  query="q"
>
  <ButtonDropdown
    isOpen={false}
    toggle={[Function]}
  >
    <Dropdown
      a11y={true}
      active={false}
      addonType={false}
      direction="down"
      group={true}
      inNavbar={false}
      isOpen={false}
      nav={false}
      setActiveFromChild={false}
      toggle={[Function]}
    >
      <Manager>
        <div
          className="btn-group"
          onKeyDown={[Function]}
        >
          <DropdownToggle
            aria-haspopup={true}
            caret={true}
            color="secondary"
          >
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <Button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle"
                  color="secondary"
                  innerRef={[Function]}
                  onClick={[Function]}
                  tag="button"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-label={null}
                    className="dropdown-toggle btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Category
                  </button>
                </Button>
              </InnerReference>
            </Reference>
          </DropdownToggle>
          <DropdownMenu
            flip={true}
            right={true}
            tag="div"
          >
            <div
              aria-hidden={true}
              className="dropdown-menu dropdown-menu-right"
              role="menu"
              tabIndex="-1"
            >
              <DropdownItem
                disabled={false}
                key="Programming languages"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Programming languages
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="Linters"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Linters
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="Code analysis"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Code analysis
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="External services"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  External services
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="Reports and stats"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Reports and stats
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="Other"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Other
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="Demos"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Demos
                </button>
              </DropdownItem>
            </div>
          </DropdownMenu>
        </div>
      </Manager>
    </Dropdown>
  </ButtonDropdown>
   
  <ButtonDropdown
    isOpen={false}
    toggle={[Function]}
  >
    <Dropdown
      a11y={true}
      active={false}
      addonType={false}
      direction="down"
      group={true}
      inNavbar={false}
      isOpen={false}
      nav={false}
      setActiveFromChild={false}
      toggle={[Function]}
    >
      <Manager>
        <div
          className="btn-group"
          onKeyDown={[Function]}
        >
          <DropdownToggle
            aria-haspopup={true}
            caret={true}
            color="secondary"
          >
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <Button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle"
                  color="secondary"
                  innerRef={[Function]}
                  onClick={[Function]}
                  tag="button"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-label={null}
                    className="dropdown-toggle btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Options
                  </button>
                </Button>
              </InnerReference>
            </Reference>
          </DropdownToggle>
          <DropdownMenu
            flip={true}
            right={true}
            tag="div"
          >
            <div
              aria-hidden={true}
              className="dropdown-menu dropdown-menu-right"
              role="menu"
              tabIndex="-1"
            >
              <DropdownItem
                disabled={false}
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Show enabled extensions
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Show disabled extensions
                </button>
              </DropdownItem>
            </div>
          </DropdownMenu>
        </div>
      </Manager>
    </Dropdown>
  </ButtonDropdown>
</ExtensionsQueryInputToolbar>
`;

exports[`ExtensionsQueryInputToolbar shows category in query as selected 1`] = `
<ExtensionsQueryInputToolbar
  onQueryChange={[Function]}
  query="category:\\"Programming languages\\""
>
  <ButtonDropdown
    isOpen={false}
    toggle={[Function]}
  >
    <Dropdown
      a11y={true}
      active={false}
      addonType={false}
      direction="down"
      group={true}
      inNavbar={false}
      isOpen={false}
      nav={false}
      setActiveFromChild={false}
      toggle={[Function]}
    >
      <Manager>
        <div
          className="btn-group"
          onKeyDown={[Function]}
        >
          <DropdownToggle
            aria-haspopup={true}
            caret={true}
            color="secondary"
          >
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <Button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle"
                  color="secondary"
                  innerRef={[Function]}
                  onClick={[Function]}
                  tag="button"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-label={null}
                    className="dropdown-toggle btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Category
                  </button>
                </Button>
              </InnerReference>
            </Reference>
          </DropdownToggle>
          <DropdownMenu
            flip={true}
            right={true}
            tag="div"
          >
            <div
              aria-hidden={true}
              className="dropdown-menu dropdown-menu-right"
              role="menu"
              tabIndex="-1"
            >
              <DropdownItem
                disabled={true}
                key="Programming languages"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="disabled dropdown-item"
                  disabled={true}
                  onClick={[Function]}
                  tabIndex="-1"
                  type="button"
                >
                  Programming languages
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="Linters"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Linters
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="Code analysis"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Code analysis
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="External services"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  External services
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="Reports and stats"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Reports and stats
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="Other"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Other
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                key="Demos"
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Demos
                </button>
              </DropdownItem>
            </div>
          </DropdownMenu>
        </div>
      </Manager>
    </Dropdown>
  </ButtonDropdown>
   
  <ButtonDropdown
    isOpen={false}
    toggle={[Function]}
  >
    <Dropdown
      a11y={true}
      active={false}
      addonType={false}
      direction="down"
      group={true}
      inNavbar={false}
      isOpen={false}
      nav={false}
      setActiveFromChild={false}
      toggle={[Function]}
    >
      <Manager>
        <div
          className="btn-group"
          onKeyDown={[Function]}
        >
          <DropdownToggle
            aria-haspopup={true}
            caret={true}
            color="secondary"
          >
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <Button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle"
                  color="secondary"
                  innerRef={[Function]}
                  onClick={[Function]}
                  tag="button"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-label={null}
                    className="dropdown-toggle btn btn-secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    Options
                  </button>
                </Button>
              </InnerReference>
            </Reference>
          </DropdownToggle>
          <DropdownMenu
            flip={true}
            right={true}
            tag="div"
          >
            <div
              aria-hidden={true}
              className="dropdown-menu dropdown-menu-right"
              role="menu"
              tabIndex="-1"
            >
              <DropdownItem
                disabled={false}
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Show enabled extensions
                </button>
              </DropdownItem>
              <DropdownItem
                disabled={false}
                onClick={[Function]}
                tag="button"
                toggle={true}
              >
                <button
                  className="dropdown-item"
                  disabled={false}
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                  type="button"
                >
                  Show disabled extensions
                </button>
              </DropdownItem>
            </div>
          </DropdownMenu>
        </div>
      </Manager>
    </Dropdown>
  </ButtonDropdown>
</ExtensionsQueryInputToolbar>
`;
